integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableAddColumnCommand.scala

```diff
@@ -93,11 +93,17 @@ private[sql] case class CarbonAlterTableAddColumnCommand(
       schemaEvolutionEntry.setAdded(newCols.toList.asJava)
       val thriftTable = schemaConverter
         .fromWrapperToExternalTableInfo(wrapperTableInfo, dbName, tableName)
+      // carbon columns based on schema order
+      val carbonColumns = carbonTable.getCreateOrderColumn(carbonTable.getTableName).asScala
+        .collect { case carbonColumn => carbonColumn.getColumnSchema }
+        .filter(!_.isInvisible)
```

Comment:
Move filter operation in collect operation by adding if clause in the case statement